projects/elfutils/Dockerfile

```diff
@@ -14,7 +14,10 @@
 #
 ################################################################################
 
-FROM gcr.io/oss-fuzz-base/base-builder
+FROM gcr.io/oss-fuzz-base/base-builder@sha256:56905c98ae0083d14da0e7371184e694560a74750533f321ac0e9145af0e8d2e
+# ! This project was pinned after a clang bump. Please remove the pin, Try to fix any build warnings and errors, as well as runtime errors
+# see https://github.com/google/oss-fuzz/pull/12365
```

Comment:
I'm not sure it's the best way to notify projects. Though given that the latest build failure wasn't reported and the primary contact pointing to the elfutils mailing doesn't seem to work anymore bug reports are likely to go unnoticed as well.

Either way elftuils is built with `clang version 18.1.8 (Fedora 18.1.8-1.fc40)` shipped by Fedora just fine. Could it be that something else is broken in the OSS-Fuzz images? The backtrace I saw points to `#include <iostream>` and that's weird.

Reply:
> that was asked instead of letting the projects fail because of warnings or such

I agree that it's better than the alternatives.

Now that the image is rolled out it looks like the failure is flaky:
https://github.com/google/oss-fuzz/actions/runs/11145617408 (where the coverage build failed)
https://github.com/google/oss-fuzz/actions/runs/11145449506 (where the coverage and UBSan builds failed)

I reproduced it more or less reliably in https://github.com/google/oss-fuzz/actions/runs/11146061465 (where all the relevant builds failed) by removing `-j` from `make`. The backtrace seems to point to `#include <iostream>` and I'm not sure why yet.

Reply:
I'm 99% sure that it's an OSS-Fuzz toolchain bug. I was curious where `stack` came from and it turns out it's from `src/.deps/srcfiles.Tpo` (which is used when `srcfiles.cxx` is compiled):
```sh
$ grep stack src/.deps/srcfiles.Tpo 
  /usr/local/bin/../include/c++/v1/deque stack \
stack:
```
and it should probably be `/usr/local/bin/../include/c++/v1/stack`. I'm guessing `elfutils` is unlucky in that it has the `stack` binary (which is rejected with all those UTF-8 warnings because it is a binary).